src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -717,5 +704,10 @@ shard_name(PG_FUNCTION_ARGS)
 	}
 
 	AppendShardIdToName(&relationName, shardId);
-	PG_RETURN_TEXT_P(cstring_to_text(relationName));
+
+	schemaId = get_rel_namespace(relationId);
```

Comment:
I haven't used `ConstructQualifiedShardName() `since it prevents us from creating a shard name on the fly (i.e. without existing shard interval for the given id). We have sample regression test for this case in `multi_name_lengths`.